sdk/common/src/main/scala/org/typelevel/otel4s/sdk/Resource.scala

```diff
@@ -117,12 +149,28 @@ object Resource {
     * @return
     *   a [[Resource]].
     */
-  val Default: Resource = TelemetrySdk.mergeIntoUnsafe(Mandatory)
+  val Default: Resource = TelemetrySdk.mergeUnsafe(Mandatory)
+
+  private final case class Impl(
+      attributes: Attributes,
+      schemaUrl: Option[String]
+  ) extends Resource {
+
+    def merge(other: Resource): Either[ResourceInitializationError, Resource] =
+      if (other == Resource.Empty) Right(this)
```

Comment:
what if `this` is `Resource.Empty`?

Reply:
Good catch. We can use a shortcut in this case too.